src/lib/ProjectM36/Client/Simple.hs

```diff
@@ -0,0 +1,98 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+module ProjectM36.Client.Simple (
+  simpleConnectProjectM36,
+  withTransaction,
+  execute,
+  query,
+  rollback,
+  close,
+  Atom(..),
+  AtomType(..),
+  DBError(..),
+  Attribute(..),
+  C.ConnectionInfo(..),
+  C.PersistenceStrategy(..),
+  C.NotificationCallback,
+  C.emptyNotificationCallback,
+  C.DatabaseContextExpr(..),
+  C.RelationalExprBase(..)  
+  ) where
+-- | A simplified client interface for Project:M36 database access.
+import ProjectM36.Error
+import ProjectM36.Base
+import qualified ProjectM36.Client as C
+import Control.Monad.Reader
+import Control.Exception.Base
+
+type DBConn = (C.SessionId, C.Connection)
+
+newtype DB a = DB {runDB :: ReaderT (C.SessionId, C.Connection) IO a}
+  deriving (Functor, Applicative, Monad)
```

Comment:
We could add `MonadIO` here without any danger. That's actually a pretty nice feature of Project-M36 that your transactions can still be atomic even with effects (because of auto-merge-to-head).